src/mame/drivers/samcoupe.cpp

```diff
@@ -458,6 +462,24 @@ static INPUT_PORTS_START( samcoupe )
 	PORT_CONFNAME(0x01, 0x00, "Real Time Clock")
 	PORT_CONFSETTING(   0x00, DEF_STR(None))
 	PORT_CONFSETTING(   0x01, "SAMBUS")
+	
+	/* Sam Coupe has single 9-pin ATARI-compatible connector but supports 2 joysticks via a splitter,
+	   this works by using a different ground for each stick (pin 8: stick 1 gnd, pin 9: stick 2 gnd.)
+	   Joysticks overlay number keys 6-0 for the stick 1 and 1-5 for stick 2 (same scheme as ZX Spectrum) */
+	
+	PORT_START("joy_1")
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1)        PORT_PLAYER(1) PORT_CODE(JOYCODE_BUTTON1)
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_UP)    PORT_8WAY PORT_PLAYER(1) PORT_CODE(JOYCODE_Y_UP_SWITCH)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN)  PORT_8WAY PORT_PLAYER(1) PORT_CODE(JOYCODE_Y_DOWN_SWITCH)
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT) PORT_8WAY PORT_PLAYER(1) PORT_CODE(JOYCODE_X_RIGHT_SWITCH)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT)  PORT_8WAY PORT_PLAYER(1) PORT_CODE(JOYCODE_X_LEFT_SWITCH)
+	
+	PORT_START("joy_2")
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT)  PORT_8WAY PORT_PLAYER(2) PORT_CODE(JOYCODE_X_LEFT_SWITCH)
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT) PORT_8WAY PORT_PLAYER(2) PORT_CODE(JOYCODE_X_RIGHT_SWITCH)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN)  PORT_8WAY PORT_PLAYER(2) PORT_CODE(JOYCODE_Y_DOWN_SWITCH)
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_UP)    PORT_8WAY PORT_PLAYER(2) PORT_CODE(JOYCODE_Y_UP_SWITCH)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_BUTTON1)        PORT_PLAYER(2) PORT_CODE(JOYCODE_BUTTON1)
```

Comment:
Please don't add `PORT_CODE` for this stuff - it prevents defaults from being applied.  They're also in conflict with each other here.  Some people actually configure their player input defaults and expect them to apply when starting a system.

Reply:
That was just cut-and-pasted from the Spectrum driver, I assumed it was being used for that very reason (preventing defaults from being applied) as in ensuring the ports were mapped to a host joystick by default rather than keyboard. The problem is the joystick ports on Sam/Spectrum just overlay the keyboard, so on real h/w moving the joystick(s) prints numbers at the BASIC prompt. So say Mame maps the joystick to keyboard keys (the default without the PORT_CODE(JOYCODE...), say Joystick 2 defaults to 'a' key, when the user hits 'a' they get a 4 printed on screen instead of an 'a' character.
Now there may well be a different way to deal with this that I'm unaware of. I also don't like the conflict but though it better than allow a conflict with the keyboard. What I was really looking for was some kind of "None" PORT_CODE whereby the joysticks wouldn't be assigned anything at all by default unless the user intervened and mapped them to whatever they wanted.
I guess it boils down to what sort of default behaviour do we strive for? do we just assume the user will sort out the mapping themselves?

Reply:
We assume drivers will respect the default mappings, be they our actual defaults or what the user sets in the "Inputs (General)" menu where you can remap things for all drivers at once.

Reply:
OK, thanks, JOYCODE removed.
So on my machine (I have one joystick), Sam joystick 2 auto-maps to R,F,D,G,A keys, which means at BASIC prompt, pressing those keys prints 4,3,1,2,5. Is there anything I can do about that? (I don't really like just assuming the user will figure it out and map joystick 2 to 'none' in the ui)

Reply:
That's pretty normal.  When you plug a mouse in on Apple II or Amiga drivers it maps over the keyboard as well.  People work it out.